From d2ff43f8fb7251ff2faba30fb8f3864f4f8c667c Mon Sep 17 00:00:00 2001
From: Tim Foster <tim.foster@joyent.com>
Date: Fri, 23 Aug 2019 13:17:29 +0100
Subject: [PATCH] OS-7967 unable to build vtfontcvt on older PIs due to queue.h
 changes Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com> Approved by:
 Jerry Jelinek <jerry.jelinek@joyent.com>

---
 usr/src/tools/vtfontcvt/Makefile | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/usr/src/tools/vtfontcvt/Makefile b/usr/src/tools/vtfontcvt/Makefile
index b8caefdb90..39c368b9a1 100644
--- a/usr/src/tools/vtfontcvt/Makefile
+++ b/usr/src/tools/vtfontcvt/Makefile
@@ -11,6 +11,7 @@
 
 #
 # Copyright 2017 Toomas Soome <tsoome@me.com>
+# Copyright 2019 Joyent, Inc.
 #
 
 CMDDIR=		$(SRC)/cmd/vtfontcvt
@@ -31,7 +32,13 @@ $(PROG):	$(OBJS)
 		$(LINK.c) $(OBJS) -o $@ $(LDLIBS)
 		$(POST_PROCESS)
 
+#
+# In order to build on older platforms, we specifically use the
+# source-tree version of queue.h. When building as part of
+# $SRC/cmd, we'll be using the $PROTO copy of queue.h anyway,
+# so this is a $SRC/tools workaround only.
+#
 %.o:		$(CMDDIR)/%.c
-		$(COMPILE.c) -o $@ $<
+		$(COMPILE.c) -I$(SRC)/uts/common -o $@ $<
 
 include		../Makefile.targ
-- 
2.21.0

